### PR TITLE
Update Scenario2

### PR DESCRIPTION
**Pull request by chalitha87**
## Purpose
This PR contains 
Create the Folder structure which relevant to the new structure of scenario .
Update the artifact.xml relevant to the new  structure.
Create a synapseConfig project and the car app which relevant to the new structure.
Update the car app deployment method in HttpMessageWithoutAlteringDownstreamTest.java class.
Update the endpoint and proxy urls.

### CODE REVIEW

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/artifact.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?><artifacts>
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/artifact.xml]: Shall we add the license header?

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/endpoints/HttpMessageWithoutAlteringDownstreamTestAddressEP.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/endpoints/HttpMessageWithoutAlteringDownstreamTestAddressEP.xml]: Shall we add the license header? Shall we update other files as well?

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/pom.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    <artifactId>02-synapseConfigProject</artifactId>
+    <name>Synapse config project for scenario 02</name>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>02-synapseConfig</module>
+        <module>02-synapseConfigCompositeApplication</module>
+    </modules>
+
+
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/pom.xml]: Unnecessary new line.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java [hunk 4]
```diff
@@ -38,12 +38,17 @@
  * downstream without altering the content
  */
 public class HttpMessageWithoutAlteringDownstreamTest extends ScenarioTestBase{
-    private final String carFileName = "2_1_1_HttpMessageWithoutAlteringDownstreamTestCompositeApplication_1.0.0";
+   // private final String carFileName = "2_1_1_HttpMessageWithoutAlteringDownstreamTestCompositeApplication_1.0.0";
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Any reason to comment this? If this line is not necessary shall we remove the line without commenting?

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java [hunk 5]
```diff
@@ -38,12 +38,17 @@
...
 
-    @BeforeClass(alwaysRun = true)
+    /*@BeforeClass(alwaysRun = true)
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: Shall we remove this block?

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/resources/testng.xml [hunk 6]
```diff
@@ -18,6 +18,15 @@
...
 <suite name="ESBTestSuite" parallel="false">
+    <listeners>
+        <!-- TestPrepExecutionListener : TestNG execution listener for perform test execution preparation -->
+        <listener class-name="org.wso2.carbon.esb.scenario.test.common.testng.listeners.TestPrepExecutionListener"/>
+
+        <!-- TestLoggingListener : Test listener to log test execution events -->
+        <listener class-name="org.wso2.carbon.esb.scenario.test.common.testng.listeners.TestLoggingListener"/>
+    </listeners>
+
+
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/resources/testng.xml]: unnecessary new line.

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/pom.xml [hunk 7]
```diff
@@ -32,9 +32,126 @@
...
+                                </property>
+                            </systemProperties>
+                            <workingDirectory>${basedir}/target</workingDirectory>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>profile_490</id>
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/pom.xml]: Do we need to maintain a separate profile for EI 4.9.0 in scenario 2 as well?

Comment (milindaperera) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/pom.xml]: no need, let's remove

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/artifact.xml (hunk 1 above)
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/artifact.xml]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/pom.xml (hunk 7 above)
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/pom.xml]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/resources/testng.xml (hunk 6 above)
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/resources/testng.xml]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 5 above)
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java (hunk 4 above)
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.1-HTTP-to-HTTP-protocol-translation/2.1.1-http-to-http-passthrough/src/test/java/org/wso2/carbon/esb/scenario/test/HttpMessageWithoutAlteringDownstreamTest.java]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/pom.xml (hunk 3 above)
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/pom.xml]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/proxy-services/2_1_1_1_HttpMessageWithoutAlteringDownstreamTestProxy.xml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<proxy name="2_1_1_1_HttpMessageWithoutAlteringDownstreamTestProxy" startOnLoad="true" transports="http" xmlns="http://ws.apache.org/ns/synapse">
```
[the remark this answers is not in the document]
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/proxy-services/2_1_1_1_HttpMessageWithoutAlteringDownstreamTestProxy.xml]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/endpoints/HttpMessageWithoutAlteringDownstreamTestAddressEP.xml [hunk 9]
```diff
@@ -0,0 +1,4 @@
...
+<endpoint name="HttpMessageWithoutAlteringDownstreamTestAddressEP" xmlns="http://ws.apache.org/ns/synapse">
```
[the remark this answers is not in the document]
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/endpoints/HttpMessageWithoutAlteringDownstreamTestAddressEP.xml]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/endpoints/HttpMessageWithoutAlteringDownstreamTestHttpEP.xml [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/endpoints/HttpMessageWithoutAlteringDownstreamTestHttpEP.xml]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/endpoints/HttpMessageWithoutAlteringDownstreamTestAddressEP.xml (hunk 2 above)
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/src/main/synapse-config/endpoints/HttpMessageWithoutAlteringDownstreamTestAddressEP.xml]: done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/artifact.xml (hunk 1 above)
Comment (chalitha87) [reply product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/artifact.xml]: Done

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.5-Switching-from-UDP-to-HTTP-or-HTTPS/02-synapseConfig/artifact.xml [hunk 11]
```diff
@@ -1,35 +1,32 @@
 <?xml version="1.0" encoding="UTF-8"?><artifacts>
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.5-Switching-from-UDP-to-HTTP-or-HTTPS/02-synapseConfig/artifact.xml]: I think we do not need this artifact.xml file. We maintain the synapse project in scenario level. Shall we remove this file?

File: product-scenarios/pom.xml [hunk 12]
```diff
@@ -74,16 +74,16 @@
...
-        <!--<module>2-Bridging-systems-that-communicate-in-different-protocols</module>-->
-        <module>3-connecting-web-apis-cloud-services</module>
-        <module>4-gateway</module>
-        <module>5-route-messages-between-systems</module>
-        <module>7-connecting-with-packaged-applications</module>
-        <module>8-connect-devices-to-enterprise</module>
-        <module>10-file-processing</module>
-        <module>13-micro-services</module>
-        <module>14-periodically-execute-an-integration-process</module>
+       <!-- <module>1-integrating-systems-that-communicate-in-heterogeneous-message-formats</module>-->
```
Comment (dulanjalidilmi) [inline product-scenarios/pom.xml]: Shall we uncomment these modules?

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/pom.xml [hunk 13]
```diff
@@ -1,13 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <modelVersion>4.0.0</modelVersion>
-  <groupId>org.wso2.ei.product-scenarios.scenario_01</groupId>
-  <artifactId>01-synapseConfig</artifactId>
+  <groupId>com.example.02-synapseConfig</groupId>
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/02-SynapseConfigProject/02-synapseConfig/pom.xml]: Shall we add correct groupId? It should be `org.wso2.ei.product-scenarios.scenario_02`

File: product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.5-Switching-from-UDP-to-HTTP-or-HTTPS/02-synapseConfig/pom.xml [hunk 14]
```diff
@@ -0,0 +1,168 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dulanjalidilmi) [inline product-scenarios/2-Bridging-systems-that-communicate-in-different-protocols/2.5-Switching-from-UDP-to-HTTP-or-HTTPS/02-synapseConfig/pom.xml]: We do not maintain synapse config project in sub-scenario level. Shall we remove this module?

File: product-scenarios/pom.xml [hunk 15]
```diff
@@ -74,16 +74,16 @@
...
+        <module>2-Bridging-systems-that-communicate-in-different-protocols</module>
+       <!-- <module>3-connecting-web-apis-cloud-services</module>-->
```
Comment (sdkottegoda) [inline product-scenarios/pom.xml]: These modules need to be uncommented

**Comment by sdkottegoda** [comment]
Closing since the changes are included in https://github.com/wso2/product-ei/pull/3232

**Pull request closed without merging**